lib/from_openstudio/geometry/room.rb

```diff
@@ -64,12 +64,166 @@ def self.properties_from_space(space)
+          unless other_eq.designLevel.empty?
+            hash[:process_loads] << Honeybee::ProcessAbridged.from_load(other_eq)
+          end
+        end
+      end
+      unless space.spaceInfiltrationDesignFlowRates.empty?
+        space.spaceInfiltrationDesignFlowRates.each do |infiltration|
+          # Only translate if flow per exterior area is specified
+          # Check if schedule exists and is of the correct type
+          if !infiltration.flowperExteriorSurfaceArea.empty? && !infiltration.schedule.empty?
+            sch = infiltration.schedule.get
+            if sch.to_ScheduleFixedInterval.is_initialized or sch.to_ScheduleRuleset.is_initialized
+              hash[:infiltration] = Honeybee::InfiltrationAbridged.from_load(infiltration)
+              break
+            end
+          end
+        end
+      end
+      unless space.designSpecificationOutdoorAir.empty?
+        hash[:ventilation] = Honeybee::VentilationAbridged.from_load(space.designSpecificationOutdoorAir.get)
+      end
+      unless space.daylightingControls.empty?
+        hash[:daylighting_control] = Honeybee::DaylightingControl.from_load(space.daylightingControls[0])
+      end
+      thermal_zone = space.thermalZone
+      unless thermal_zone.empty?
+        thermal_zone = space.thermalZone.get
+        unless thermal_zone.thermostatSetpointDualSetpoint.empty?
+          hash[:setpoint] = {}
+          thermostat = thermal_zone.thermostatSetpointDualSetpoint.get
+          hash[:setpoint][:identifier] = thermostat.nameString
+          if thermostat.heatingSetpointTemperatureSchedule.empty?
+            # if heating setpoint schedule is not specified create a new setpoint schedule and assign to HB thermostat object.
+            # first check if schedule is already created
+            if $heating_setpoint_schedule.nil?
+              openstudio_model = OpenStudio::Model::Model.new
+              openstudio_schedule = OpenStudio::Model::ScheduleRuleset.new(openstudio_model)
+              openstudio_schedule.setName('Heating Schedule Default')
+              openstudio_sch_type_lim = OpenStudio::Model::ScheduleTypeLimits.new(openstudio_model)
+              openstudio_sch_type_lim.setName('Temperature')
+              openstudio_sch_type_lim.setNumericType('Temperature')
+              openstudio_schedule.defaultDaySchedule.setName('Heating Day Default')
+              openstudio_schedule.defaultDaySchedule.addValue(OpenStudio::Time.new(0,24,0,0), 100)
+              openstudio_schedule.defaultDaySchedule.setScheduleTypeLimits(openstudio_sch_type_lim)
+              $heating_setpoint_schedule = Honeybee::ScheduleRulesetAbridged.from_schedule_ruleset(openstudio_schedule)
+            end
+            hash[:setpoint][:heating_schedule] = $heating_setpoint_schedule[:identifier]
+          else
+            heating_schedule = thermostat.heatingSetpointTemperatureSchedule.get
+            hash[:setpoint][:heating_schedule] = heating_schedule.nameString
```

Comment:
We need the `sch.to_ScheduleFixedInterval.is_initialized or sch.to_ScheduleRuleset.is_initialized` check here.